parsons/redash/redash.py

```diff
@@ -0,0 +1,168 @@
+import json
+import logging
+import requests
+import time
+
+from parsons.etl.table import Table
+
+logger = logging.getLogger(__name__)
+
+
+class Redash(object):
+    """
+    Instantiate Redash Class
+
+    `Args:`
+        base_url: str
+            The base url for your redash instance (excluding the final /)
+        user_api_key: str
+            The user API key found in the User's profile screen
+        pause_time int
+            Specify time between polling for refreshed queries (Defaults to 3 seconds)
+        verify: bool
+            For https requests, should the certificate be verified (Defaults to True)
+    `Returns:`
+        Redash Class
+    """
+    args = {'base_url': None,
+            'user_api_key': None,
+            'pause_time': 3,
+            'verify': True,
+            'query_id': None,
+            'params': None}
+
+    def __init__(self,
+                 base_url="http://localhost",
+                 user_api_key=None,
+                 pause_time=3,
+                 verify=True):
+        self.base_url = base_url
+        self.user_api_key = user_api_key
+        self.pause = pause_time   # seconds to delay
+        self.verify = verify  # for https requests
+        self.session = requests.Session()
+        if user_api_key:
+            self.session.headers.update({'Authorization': f'Key {user_api_key}'})
+
+    def _poll_job(self, session, job, query_id):
+        while job['status'] not in (3, 4):
+            poll_url = '{}/api/jobs/{}'.format(self.base_url, job['id'])
+            response = session.get(poll_url, verify=self.verify)
+            response_json = response.json()
+            job = response_json.get(
+                'job',
+                {'status': 'Error NO JOB IN RESPONSE: {}'.format(json.dumps(response_json))})
+            logger.debug('poll', poll_url, query_id, job['status'], job.get('error'))
+            time.sleep(self.pause)
+
+        if job['status'] == 3:  # 3 = completed
+            return job['query_result_id']
+        elif job['status'] == 4:  # 3 = ERROR
+            raise Exception('Redash Query {} failed: {}'.format(query_id, job['error']))
+
+    def get_fresh_query_results(self, query_id, params=None):
+        """
+        Make a fresh query result and get back the CSV http response object back
+        with the CSV string in result.content
+
+        `Args:`
+            query_id: str or int
+                The query id of the query
+            params: dict
```

Comment:
I think the more "Parson-ic" (see what I did there?) way of doing this would be to leverage `**kwargs`. If possible, we also like to drop a link in the docstring to the corresponding API docs on possible args.. though a quick scan of the Redash docs makes me think such a link might not exist.

Reply:
Also, if there are very common parameters that get passed in (not sure, since I'm not familiar with Redash), it might be a good idea to name them explicitly as optional function args.